server/routes/api/stats.py

```diff
@@ -408,3 +411,172 @@ def get_place_stat_date_within_place():
+                elif curr_sv_date < curr_date or curr_sv_date.startswith(
+                        curr_date):
+                    curr_date = curr_sv_date
+            have_data = False
+            place_date_row = [place, place_name]
+            for sv, idx in sv_curr_index.items():
+                # if a sv doesn't have any more data left, just append empty cells
+                if idx >= len(sv_dates[sv]):
+                    place_date_row.extend(["", "", ""])
+                    continue
+                curr_sv_date = sv_dates[sv][idx]
+                # Add data for an sv if the current date to add for that sv is
+                # equal to or encompassing the chosen date. Eg. if the chosen
+                # date is 2015-01-02, then we can add data from 2015, 2015-01 or
+                # 2015-01-02.
+                if curr_date.startswith(curr_sv_date):
+                    value = stat_set_series.get(place, {}).get("data", {}).get(
+                        sv, {}).get("val", {}).get(curr_sv_date, "")
+                    place_date_row.extend(
+                        [curr_sv_date, value,
+                         sv_source.get(sv, "")])
+                    sv_curr_index[sv] += 1
+                else:
+                    place_date_row.extend(["", "", ""])
+                have_data = have_data or sv_curr_index[sv] < len(sv_dates[sv])
+            result.append(place_date_row)
+    return result
+
+
+@bp.route('/api/stats/csv/within-place')
+@cache.cached(timeout=3600 * 24, query_string=True)
+def get_stats_within_place_csv():
+    """Gets the statistical variable data as a csv for child places of a
+    certain place type contained in a parent place. If no date range specified,
+    gets data for all dates of a series. "latest" will get the latest date data.
+    """
+    parent_place = request.args.get('parentPlace')
+    if not parent_place:
+        return Response(json.dumps('error: must provide a parentPlace field'),
+                        400,
+                        mimetype='application/json')
+    child_type = request.args.get('childType')
+    if not child_type:
+        return Response(json.dumps('error: must provide a childType field'),
+                        400,
+                        mimetype='application/json')
+    stat_vars = request.args.getlist('statVars')
+    if not stat_vars:
+        return Response(json.dumps('error: must provide a statVars field'),
+                        400,
+                        mimetype='application/json')
+    min_date = request.args.get('minDate')
+    max_date = request.args.get('maxDate')
+    row_limit = request.args.get('rowLimit')
+    if row_limit:
+        row_limit = int(row_limit)
+    result_csv = []
+    header_row = ["placeDcid", "placeName"]
+    for sv in stat_vars:
+        header_row.extend(["Date:" + sv, "Value:" + sv, "Source:" + sv])
+    result_csv.append(header_row)
+    if min_date and max_date and min_date == max_date:
```

Comment:
in any case where min_date == max_date, we want to go in here because we are getting data for a single date instead of a series of dates .. added some documentation to clarify